File: easy/python/longest_palindrome.py
from typing import Dict


class Solution:
    def longestPalindrome(self, s: str) -> int:
        char_count: Dict[str, int] = {}

        for char in s:
            if char in char_count:
                char_count[char] += 1
            else:
                char_count[char] = 1

        len = 0
        has_odd = False
        for count in char_count.values():
            if count % 2 == 0:
                len += count
            else:
                len += count - 1
                has_odd = True

        if has_odd:
            len += 1
        return len

File: easy/ruby/longest_palindrome.rb
# @param {String} s
# @return {Integer}
def longest_palindrome(s)
  hash = Hash.new(0)
  s.each_char do |char|
    hash[char] += 1
  end

  p_length = 0
  has_odd = false
  hash.each do |count|
    if count[1].even?
      p_length += count[1]
    else
      p_length += count[1] - 1
      has_odd = true
    end
  end
  p_length += 1 if has_odd
  p_length
end
